com.onesignal.unity.core/Runtime/Notifications/Models/INotification.cs

```diff
@@ -39,165 +39,165 @@ public enum LockScreenVisibility {
-        public string id;
-        public string text;
-        public string icon;
+    public interface IActionButton {
+        string Id { get; }
+        string Text { get; }
+        string Icon { get; }
     }
 
     /// <summary>
     /// If a background image was set, this object will be available
     /// </summary>
     /// <remarks>Android only</remarks>
-    [Serializable] public sealed class BackgroundImageLayout {
-        public string image;
-        public string titleTextColor;
-        public string bodyTextColor;
+    public interface IBackgroundImageLayout {
+        public string Image { get; }
+        public string TitleTextColor { get; }
+        public string BodyTextColor { get; }
     }
     
     /// <summary>
     /// See full documentation at
     /// https://documentation.onesignal.com/docs/sdk-notification-event-handlers#osnotification-class
     /// </summary>
-    [Serializable] public class NotificationBase {
+    public interface INotificationBase {
         /// <summary>
         /// OneSignal notification UUID
         /// </summary>
-        public string notificationId;
+        string NotificationId { get; }
         
         /// <summary>
         /// Name of Template from <a href="https://documentation.onesignal.com/docs/templates">Templates</a>
         /// </summary>
-        public string templateName;
+        string TemplateName { get; }
         
         /// <summary>
         /// Unique Template Identifier from <a href="https://documentation.onesignal.com/docs/templates">Templates</a>
         /// </summary>
-        public string templateId;
+        string TemplateId { get; }
         
         /// <summary>
         /// Title text of the notification
         /// </summary>
-        public string title;
+        string Title { get; }
         
         /// <summary>
         /// Body text of the notification
         /// </summary>
-        public string body;
+        string Body { get; }
 
         /// <summary>
         /// URL opened when opening the notification
         /// </summary>
-        public string launchURL;
+        string LaunchURL { get; }
         
         /// <summary>
         /// Sound resource played when the notification is shown
         /// https://documentation.onesignal.com/docs/customize-notification-sounds
         /// </summary>
-        public string sound;
+        string Sound { get; }
 
         /// <summary>
         /// Collapse id for the notification
         /// </summary>
-        public string collapseId;
+        string CollapseId { get; }
         
         /// <summary>
         /// Gets custom additional data that was sent with the notification. Set on the dashboard under
         /// Options > Additional Data or with the data field on the REST API.
         /// </summary>
-        public Dictionary<string, object> additionalData;
+        Dictionary<string, object> AdditionalData { get; }

```

Comment:
you might be able to make this `IDictionary<string, object>` to provide a little more abstraction